src/core/models/payloads/tagged_data.c

```diff
@@ -0,0 +1,264 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdio.h>
+#include <string.h>
+
+#include "core/models/message.h"
+#include "core/models/payloads/tagged_data.h"
+#include "core/utils/macros.h"
+
+tagged_data_t *tagged_data_new() {
+  tagged_data_t *tagged_data = malloc(sizeof(tagged_data_t));
+  if (tagged_data) {
+    tagged_data->tag = NULL;
+    tagged_data->data = NULL;
+  }
+  return tagged_data;
+}
+
+void tagged_data_free(tagged_data_t *tagged_data) {
+  if (tagged_data) {
+    if (tagged_data->tag) {
+      byte_buf_free(tagged_data->tag);
+    }
+    if (tagged_data->data) {
+      byte_buf_free(tagged_data->data);
+    }
```

Comment:
the point is checked in `byte_buf_free` too.
```suggestion
    byte_buf_free(tagged_data->tag);
    byte_buf_free(tagged_data->data);
```

Reply:
Thanks. It's fixed.